numpy/random/mtrand/mtrand.pyx

```diff
@@ -1114,15 +1114,15 @@ cdef class RandomState:
                 # __index__ must return an integer by python rules.
                 pop_size = operator.index(a.item())
             except TypeError:
-                raise ValueError("a must be 1-dimensional or an integer")
-            if pop_size <= 0:
-                raise ValueError("a must be greater than 0")
+                raise ValueError("'a' must be 1-dimensional or an integer")
+            if pop_size <= 0 and np.prod(size) != 0:
```

Comment:
a bit unintuitive that this works for `None`, but I guess OK, could also add `size is None` explicitly.

Reply:
That also has the advantage that `np.prod` probably adds a bit of overhead, but maybe negligible in any case.